pkg/crawler/central/central.go

```diff
@@ -0,0 +1,376 @@
+package central
+
+import (
+	"cmp"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"log/slog"
+	"net/http"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync/atomic"
+	"unicode"
+
+	"github.com/elireisman/maven-index-reader-go/pkg/config"
+	"github.com/elireisman/maven-index-reader-go/pkg/data"
+	"github.com/elireisman/maven-index-reader-go/pkg/readers"
+	"github.com/hashicorp/go-retryablehttp"
+	"github.com/samber/lo"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy-java-db/pkg/crawler/central/maven"
+	"github.com/aquasecurity/trivy-java-db/pkg/crawler/sha1"
+	"github.com/aquasecurity/trivy-java-db/pkg/crawler/types"
+	"github.com/aquasecurity/trivy-java-db/pkg/hash"
+	"github.com/aquasecurity/trivy-java-db/pkg/index"
+)
+
+var _ types.Source = (*Source)(nil)
```

Comment:
IIUC you check that `Source` implements `types.Source`. (same for `gcs.go`)

But do we need that now?
Or is there a meaning to this that I'm missing?

Reply:
It's not strictly necessary, but I keep it because it helps quickly notice when it no longer satisfies the `types.Source` interface, and it also makes it clear that this interface is being implemented. I'm fine with removing it if you want.

Reply:
I just decided to clarify that I understood correctly why this is necessary :)
it doesn't matter to me, act at your own discretion.